src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -0,0 +1,495 @@
+                {
+                    walkTo = null;
+                    return COOK_GET_FIREWOOD;
+                }
+
+                InventoryUtils.transferXOfFirstSlotInProviderWithIntoInItemHandler(
+                        new InvWrapper(worker.getInventoryCitizen()), TileEntityFurnace::isItemFuel, Constants.STACKSIZE,
+                        new InvWrapper((TileEntityFurnace) entity), FUEL_SLOT);
+            }
+
+            ((BuildingCook) getOwnBuilding()).setIsSomethingInOven(true);
+            walkTo = null;
+            return START_WORKING;
+        }
+        walkTo = null;
+        setDelay(STANDARD_DELAY);
+        return COOK_COOK_FOOD;
+    }
+
+    /**
+     * Try to transfer a item matching a predicate from a position to the cook.
+     *
+     * @param pos       the position to transfer it from.
+     * @param predicate the predicate to evaluate.
+     * @return true if succesful.
+     */
+    private boolean tryTransferFromPosToCook(final BlockPos pos, @NotNull final Predicate<ItemStack> predicate)
+    {
+        final TileEntity entity = world.getTileEntity(pos);
+        if (entity instanceof TileEntityChest)
+        {
+            return InventoryUtils.transferXOfFirstSlotInProviderWithIntoNextFreeSlotInItemHandler(
+                    new InvWrapper((TileEntityChest) entity),
+                    predicate,
+                    Constants.STACKSIZE,
+                    new InvWrapper(worker.getInventoryCitizen()));
+        }
+        else if (entity instanceof TileEntityRack)
+        {
+            return InventoryUtils.transferXOfFirstSlotInProviderWithIntoNextFreeSlotInItemHandler(
+                    ((TileEntityRack) entity).getInventory(),
+                    predicate,
+                    Constants.STACKSIZE,
+                    new InvWrapper(worker.getInventoryCitizen()));
+        }
+        return false;
+    }
+
+    private BlockPos getPositionOfOvenToRetrieveFrom()
+    {
+        for (final BlockPos pos : ((BuildingCook) getOwnBuilding()).getFurnaces())
+        {
+            final TileEntity entity = world.getTileEntity(pos);
+            if (entity instanceof TileEntityFurnace && !((TileEntityFurnace) entity).isBurning()
+                    && (!ItemStackUtils.isEmpty(((TileEntityFurnace) entity)
+                    .getStackInSlot(RESULT_SLOT))
+                    || !ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(COOK_SLOT))))
+            {
+                worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.retrieving"));
+                return pos;
+            }
+        }
+        ((BuildingCook) getOwnBuilding()).setIsSomethingInOven(false);
+        return null;
+    }
+
+    @Override
+    protected int getActionsDoneUntilDumping()
+    {
+        return 1;
+    }
+
+    private AIState startWorking()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)